PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/ui/WebAppSettingPanel.java

```diff
@@ -610,14 +626,49 @@ private void resetWidget() {
         txtSelectedWebApp.setText("");
     }
 
-    private void toggleSlotPanel() {
-        cbDeploymentSlots.setVisible(deployToSlotCheckBox.isSelected());
-        if (!deployToSlotCheckBox.isSelected() || selectedWebApp == null) {
+    private void toggleSlotPanel(final boolean isDeployToSlot) {
+        deployToSlotCheckBox.setSelected(isDeployToSlot);
+        cbDeploymentSlots.setVisible(isDeployToSlot);
+        newSlotPanel.setVisible(isDeployToSlot && cbDeploymentSlots.getSelectedItem() == Constants.CREATE_NEW_SLOT);
+        if (!isDeployToSlot || selectedWebApp == null) {
             return;
         }
-        final WebApp app = selectedWebApp.getResource();
         cbDeploymentSlots.removeAllItems();
-        app.deploymentSlots().list().stream().forEach(slot -> cbDeploymentSlots.addItem(slot.name()));
+        slotConfigurationSourceComboBox.removeAllItems();
+        webAppDeployViewPresenter.onLoadDeploymentSlots(webAppConfiguration.getSubscriptionId(), selectedWebApp.getResource().id());
+    }
+
+    @Override
+    public void fillDeploymentSlots(@NotNull final List<DeploymentSlot> slots) {
+        cbDeploymentSlots.removeAllItems();
+        slotConfigurationSourceComboBox.removeAllItems();
+        final List<String> configurationSources = new ArrayList<String>();
+        final List<String> deploymentSlots = new ArrayList<String>();
+        configurationSources.add(Constants.DO_NOT_CLONE_SLOT_CONFIGURATION);
+        configurationSources.add(selectedWebApp.getResource().name());
+
+        slots.stream().forEach(slot -> {
+            deploymentSlots.add(slot.name());
+            configurationSources.add(slot.name());
+        });
+        deploymentSlots.add(Constants.CREATE_NEW_SLOT);
+
+        configurationSources.forEach(c -> {
+            slotConfigurationSourceComboBox.addItem(c);
+            if (Comparing.equal(c, webAppConfiguration.getNewSlotConfigurationSource())) {
+                slotConfigurationSourceComboBox.setSelectedItem(c);
+            }
+        });
+        deploymentSlots.forEach(s -> {
```

Comment:
please put these logic before `configurationSources`

Reply:
Refactor this part in PR #2178